### ISSUE TITLE
Show nested columns in Crate

### ISSUE THREAD

**Issue by mikethebeer** [metabase/metabase#2962]
In Crate it is possible to store fields of type `object`. This allows you to define nested columns. This columns are other fields of any type (that Crate supports), even further object columns.

``` sql
create table my_table12 (
      title string,
      author object as (
        name string,
        birthday timestamp
      )
);
```

Currently only the object column is displayed in filter or in the grouping field. In Crate you need to query the `information_schema.columns` table to fetch those nested columns too.

However, if a quoting is necessary on those columns this is a bit tricky. Since it is not allowed to just quote the returned column as a whole.

e.g.
correct: `"MyObj"['x']`
wrong: `"MyObj['x']"`

But it works fine if you don't need the quoting of fields at all. However, if you want to use special semantics in field-names like (umlauts, camel-case, whitespace...) you need to quote those columns in order to fetch values from them.

**Comment by salsakran** [comment]
@mikethebeer where did you and your team end up on the crate jdbc driver's getColumns function returning a string we could use directly? 

It's *really* strange to have to re-write columns names that come back from that function in order to use them.

**Comment by mikethebeer** [comment]
@salsakran we decided not to return these nested columns in the `getColumns` function to be compliant with JDBC spec. A third party application that uses `getColumns` always thinks, that it can quote the returned field-names. This is not the case with Crate since those fields needs special quoting to ensure that field-names can use camel-case names or whitespace characters in it.

**Comment by salsakran** [comment]
Is there another way we can get those nested columns? With mongo we inspect a sample of documents to figure out nesting. Can we do something similar w/ crate?

**Comment by mikethebeer** [comment]
@salsakran yes, you don't need to query nested columns in a separate way than normal fields. All of them are listed in the schemata table called `information_schema.columns`.

e.g. this list shows every column (incl. nested columns) of the table `udc_coordinates`
```sql
cr> select * from information_schema.columns where table_name = 'udc_coordinates';
+--------------------------+-----------+-----------------------+--------------+------------------+-------------+-----------------+
| column_name              | data_type | generation_expression | is_generated | ordinal_position | schema_name | table_name      |
+--------------------------+-----------+-----------------------+--------------+------------------+-------------+-----------------+
| cluster_id               | string    |                  NULL | FALSE        |                1 | doc         | udc_coordinates |
| crate_version            | string    |                  NULL | FALSE        |                2 | doc         | udc_coordinates |
| hardware_address         | string    |                  NULL | FALSE        |                3 | doc         | udc_coordinates |
| java_version             | string    |                  NULL | FALSE        |                4 | doc         | udc_coordinates |
| kernel                   | object    |                  NULL | FALSE        |                5 | doc         | udc_coordinates |
| kernel['Arch']           | string    |                  NULL | FALSE        |                6 | doc         | udc_coordinates |
| kernel['Description']    | string    |                  NULL | FALSE        |                7 | doc         | udc_coordinates |
| kernel['Machine']        | string    |                  NULL | FALSE        |                8 | doc         | udc_coordinates |
| kernel['Name']           | string    |                  NULL | FALSE        |                9 | doc         | udc_coordinates |
```

**Comment by camsaul** [comment]
Implemented by #3858
